internal/server/storage/s3/miniod/admin_client.go

```diff
@@ -0,0 +1,250 @@
+package miniod
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/google/uuid"
+
+	"github.com/lxc/incus/v6/shared/subprocess"
+)
+
+// AddServiceAccountResp is the response body of the add service account call.
+type AddServiceAccountResp struct {
+	AccessKey    string `json:"accessKey,omitempty"`
+	SecretKey    string `json:"secretKey,omitempty"`
+	SessionToken string `json:"sessionToken,omitempty"`
+}
+
+// InfoServiceAccountResp is the response body of the info service account call.
+type InfoServiceAccountResp struct {
+	AccountStatus string         `json:"accountStatus"`
+	ParentUser    string         `json:"parentUser"`
+	Policy        map[string]any `json:"policy"`
+}
+
+// AdminClient represents minio client.
+type AdminClient struct {
+	process *Process
+	alias   string
+	varPath string
+}
+
+// configDir returns path to the configuration directory for mc command.
+func (c *AdminClient) configDir() string {
+	return filepath.Join(c.varPath, "mc", "config")
+}
+
+// policyDir returns path to the location with temporary policy files.
+func (c *AdminClient) policyDir() string {
+	return filepath.Join(c.varPath, "mc", "policy")
+}
+
+// runClientCommand runs 'mc' command.
+func (c *AdminClient) runClientCommand(ctx context.Context, workDir string, extraArgs ...string) (string, error) {
+	args := []string{"-C", c.configDir()}
+
+	args = append(args, extraArgs...)
+
+	cmd := exec.CommandContext(ctx, "mc", args...)
```

Comment:
It's also common for distros to ship the MinIO Client as `mcli` -- would probably be good to add that as a fallback test if `mc` doesn't exist.

Reply:
Fixed. First we are trying `mc` then `mcli`